pkg/volume/flexvolume/plugin.go

```diff
@@ -119,13 +119,18 @@ func (plugin *flexVolumePlugin) GetVolumeName(spec *volume.Spec) (string, error)
 	call := plugin.NewDriverCall(getVolumeNameCmd)
 	call.AppendSpec(spec, plugin.host, nil)
 
-	_, err := call.Run()
-	if isCmdNotSupportedErr(err) {
-		return (*pluginDefaults)(plugin).GetVolumeName(spec)
-	} else if err != nil {
-		return "", err
+	status, err := call.Run()
+	if err != nil {
```

Comment:
The error handling here is not right. Not all the error is well handled.

For `isCmdNotSupportedErr(err) == true`, non-nil error will not be returned.

Reply:
If isCmdNotSupportedErr(err) == true, then the status =nil, so will run next line
` name, err := (*pluginDefaults)(plugin).GetVolumeName(spec)`

Reply:
imporve : `isCmdNotSupportedErr` already has `err != nil`